### ISSUE TITLE
String index out of range

### ISSUE THREAD

**Issue by grolschie** [vitalidze/chromecast-java-api-v2#112]
The following is a simplified example of my code:
```
String mUrl = "http://sc2.streamwithq.com:8054/stream";
chromecast.load(mUrl);
```
 and am getting the following error:

```
"java.lang.StringIndexOutOfBoundsException: String index out of range: -10
        at java.lang.String.substring(String.java:1975)
        at su.litvak.chromecast.api.v2.ChromeCast.load(ChromeCast.java:433)"
```

Here is the method in the API where the error is caused:
```
public final MediaStatus load(String url) throws IOException {
        return load(url.substring(url.lastIndexOf('/') + 1, url.lastIndexOf('.')), null, url, null);
    }
```

It crashes when sending radio stream URLs without a file suffix such as .mp3, i.e. on URLs of the format above.

**Comment by grolschie** [comment]
A workaround is to modify the URL from:
...
http://sc2.streamwithq.com:8054/stream
...
to:
...
http://sc2.streamwithq.com:8054/;stream.mp3
...
But this doesn't work with all URLs. Also, users could still crash the app by entering the original URL.

**Comment by vitalidze** [comment]
The method tries to guess the content type by the URL. What is the suggested way to fix this? I can see two possible solution:

1. Add a validation for the URL and throw `IllegalArgumentException` if it does not match the expected format.

2. Pass the URL as is without specifying content type hoping that chromecast receiver will handle it some how.

You are welcome to propose another solution.

**Comment by grolschie** [comment]
Hmmm... It's your call really. I would lean towards option 2, but both options have their merits.

**Comment by vitalidze** [comment]
I decided to make it less restrictive, so it will set the "stream" as the Media Title in your case. Also it will try to find out the MIME Type by connecting quickly to the specifying URL as it was done for other similar methods. The change will be included in some ongoing minor version of the library.

Thank you for your feedback.

**Comment by grolschie** [comment]
Thank you!

**Comment by vitalidze** [comment]
Fix was included in just released version 0.11.1, which will become available today. Please let me know if the problem persists.
